rye/src/lock.rs

```diff
@@ -425,13 +450,17 @@ fn generate_lockfile(
             .lockfile(
                 py_ver,
                 requirements_file_in,
+                overrides_file_in,
                 &requirements_file,
                 lock_options.pre,
                 env::var("__RYE_UV_EXCLUDE_NEWER").ok(),
                 upgrade,
                 keyring_provider,
             )?;
     } else {
+        if overrides_file_in.is_some() {
+            bail!("dependency overrides require the uv backend");
```

Comment:
can this maybe be a warning? Don't know

Reply:
it would ignore an explicit instruction to overwrite a dependency, I feel like it should be treated as an error.

Reply:
It's a question of design, I don't know, on one hand the project is a pyproject.toml file which can be interpreted by standard python tools, not just Rye. And those tools would ignore everything inside `tool.rye` anyway. Maybe error when rye is managing it is good yeah.